src/dom/Value.ts

```diff
@@ -159,9 +159,32 @@ export interface Value {
   /**
    * For the Struct, List, and SExpression types, returns an array containing the container's
    * nested values; otherwise throws an Error.
+   * If there are duplicate fields in a Struct, then it only returns the last value for that field.
    */
   elements(): Value[];
 
+  /**
+   * For the Struct, List, and SExpression types, returns an array containing the container's
+   * nested values; otherwise throws an Error.
+   * If there are duplicate fields in a Struct, then it only returns all the values for that field.
+   *
+   * Like the `elements()` method, but returns an array containing the field name/value pairs with *all* of
+   * the associated values to a field name, instead of field name/value pairs with just the last associated value
+   * to a field name. For example, for the
+   * following struct:
```

Comment:
Ah, now I understand why `Struct#allElements` was returning `Value[][]`.

The `Value#elements` method makes the most sense when you're working with a `list` or `sexp`. In that use case, it's straightforward: it gives you all of the child elements of the sequence.

For `struct`s, it gets weird. It only returns the field values. If a user wanted the name/value(s) pairs, they could use `fields()` or `allFields()` to get them. Arguably, we could've made `elements()` just throw an exception if you called it on a `struct`, but we decided to make it act more or less like the other containers.

For consistency with its behavior for `list` and `sexp`, I think this should return a flat `Value[]` with a list of the struct's field values.